core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -72,7 +72,8 @@ private[spark] class ReliableCheckpointRDD[T: ClassTag](
       .sortBy(_.toString)
     // Fail fast if input files are invalid
     inputFiles.zipWithIndex.foreach { case (path, i) =>
-      if (!path.toString.endsWith(ReliableCheckpointRDD.checkpointFileName(i))) {
+      if (path.getName != ReliableCheckpointRDD.checkpointFileName(i) &
```

Comment:
You may be right that you can assume this code won't encounter files with older shorter split numbers. But it seems easier to just look for a file name ending in "part-[0-9]+" here rather than special case an "old format"